public/Enable-DbaReplDistributor.ps1

```diff
@@ -0,0 +1,92 @@
+        By default, when something goes wrong we try to catch it, interpret it and give you a friendly warning message.
+        This avoids overwhelming you with "sea of red" exceptions, but is inconvenient because it basically disables advanced scripting.
+        Using this switch turns this "nice by default" feature off and enables you to catch exceptions with your own try/catch.
+
+    .PARAMETER WhatIf
+        If this switch is enabled, no actions are performed but informational messages will be displayed that explain what would happen if the command were to run.
+
+    .PARAMETER Confirm
+        If this switch is enabled, you will be prompted for confirmation before executing any operations that change state.
+
+    .NOTES
+        Tags: repl, Replication
+        Author: Jess Pomfret (@jpomfret), jesspomfret.com
+
+        Website: https://dbatools.io
+        Copyright: (c) 2023 by dbatools, licensed under MIT
+        License: MIT https://opensource.org/licenses/MIT
+
+    .LINK
+        https://dbatools.io/Enable-DbaReplDistributor
+
+    .EXAMPLE
+        PS C:\> Enable-DbaReplDistributor -SqlInstance mssql1
+
+        Enables distribution for the mssql1 instance.
+
+    .EXAMPLE
+        PS C:\> Enable-DbaReplDistributor -SqlInstance mssql1 -DistributionDatabase repDatabase
+
+        Enables distribution for the mssql1 instance and names the distribution database repDatabase.
+
+    #>
+    [CmdletBinding(DefaultParameterSetName = "Default", SupportsShouldProcess, ConfirmImpact = 'Medium')]
+    param (
+        [parameter(Mandatory, ValueFromPipeline)]
+        [DbaInstanceParameter[]]$SqlInstance,
+        [PSCredential]$SqlCredential,
+        [string]$DistributionDatabase = 'distribution',
+        [switch]$EnableException
+    )
+    process {
+        foreach ($instance in $SqlInstance) {
+            try {
+                $replServer = Get-DbaReplServer -SqlInstance $instance -SqlCredential $SqlCredential
+            } catch {
+                Stop-Function -Message "Failure" -Category ConnectionError -ErrorRecord $_ -Target $instance -Continue
+            }
+            Write-Message -Level Verbose -Message "Enabling replication distribution for $instance"
+
+            try {
+                if ($PSCmdlet.ShouldProcess($instance, "Enabling distributor for $instance")) {
+                    $distributionDb = New-Object Microsoft.SqlServer.Replication.DistributionDatabase
+                    $distributionDb.ConnectionContext = $replServer.ConnectionContext
+                    $distributionDb.Name = $DistributionDatabase
+
+                    #TODO: lots more properties to add as params
+                    $replServer.InstallDistributor($null, $distributionDb)
+                }
+            } catch {
+                Stop-Function -Message "Unable to enable replication distributor" -ErrorRecord $_ -Target $instance -Continue
+            }
+
+            $replServer.Refresh()
+            $replServer
```

Comment:
I would not execpt this to be returned if it was not enabled?

Reply:
agreed - moved inside try block